cli/src/pcluster/api/controllers/cluster_operations_controller.py

```diff
@@ -325,3 +398,47 @@ def _handle_config_validation_error(e: ConfigValidationError) -> CreateClusterBa
             configuration_validation_errors=config_validation_messages, message="Invalid cluster configuration"
         )
     )
+
+
+def _analyze_changes(changes):
+    if changes is None or len(changes) <= 1:
+        return [], []
+
+    change_set = []
+    errors = []
+    key_indexes = {key: index for index, key in enumerate(changes[0])}
+
+    for row in changes[1:]:
+        parameter = _get_yaml_path(row[key_indexes["param_path"]], row[key_indexes["parameter"]])
+        new_value = row[key_indexes["new value"]]
+        old_value = row[key_indexes["old value"]]
+        check_result = row[key_indexes["check"]]
+        message = _create_message(row[key_indexes["reason"]], row[key_indexes["action_needed"]])
+        if check_result != UpdatePolicy.CheckResult.SUCCEEDED:
+            errors.append(
+                UpdateError(parameter=parameter, requested_value=new_value, message=message, current_value=old_value)
+            )
+        change_set.append(Change(parameter=parameter, requested_value=new_value, current_value=old_value))
+    return change_set, errors
+
+
+def _create_message(failure_reason, action_needed):
+    message = None
+    if failure_reason:
+        message = failure_reason
+    if action_needed:
+        message = f"{message}. {action_needed}" if message else action_needed
+    return message or "Error during update"
+
+
+def _get_yaml_path(path, parameter):
```

Comment:
can we add a unit test for this?

Reply:
This is exercised through the test for the `update` public method, as the method below. Do we usually test private methods?

Reply:
Note: added `test_handling_of_change_sets` to check top level behavior against various changesets.